fastplotlib/utils/functions.py

```diff
@@ -455,17 +455,17 @@ def subsample_array(
     np.ndarray
         subsample of the input array
     """
-    if np.prod(arr.shape) <= max_size:
+    if np.prod(arr.shape, dtype=np.int64) <= max_size:
```

Comment:
I think a better way is:

```
full_shape = np.array(arr.shape, dtype=np.uint64)
```

and then use this throughout

also note the uint64 instead of int64, has a higher bound